# Chapter 4 Coding Exercises

## Algorithm Worthbench 

Question 3: 
Write a loop the displays the following set of numbers: 
0, 10, 20, 30, 40, 50 ... 1000

In [1]:
#Setting up Range for set of Numbers
for num in range(0, 1001, 10):
    #Displaying set of numbers 
    print(num)

0
10
20
30
40
50
60
70
80
90
100
110
120
130
140
150
160
170
180
190
200
210
220
230
240
250
260
270
280
290
300
310
320
330
340
350
360
370
380
390
400
410
420
430
440
450
460
470
480
490
500
510
520
530
540
550
560
570
580
590
600
610
620
630
640
650
660
670
680
690
700
710
720
730
740
750
760
770
780
790
800
810
820
830
840
850
860
870
880
890
900
910
920
930
940
950
960
970
980
990
1000


Question 2: Write a while loop that asks the user to enter two numbers. The numbers should be added and the sum displayed. The loop should ask the user if he or she wishes to preform the operation again. If so, the loop should repeat, otherwise it should termnate. 

In [2]:
#create a variable to control 
keep_going = 'y'

#Ask user for numbers to enter into while loop 
while keep_going == 'y':
    #Enter first number
    first_num = float(input('Enter First Number: '))
    #Enter second number 
    second_num = float(input('Enter Second Number: '))
    
    #Calculation (numbers should be added according to problem)
    sum = first_num + second_num

    #display the sum
    print(sum)

    # Ask if user would like to preform another calculation 
    keep_going = input('Do you wish to preform another ' + 'calculation (Enter y for yes):')

Enter First Number: 45
Enter Second Number: 65
110.0
Do you wish to preform another calculation (Enter y for yes):y
Enter First Number: 101
Enter Second Number: 2
103.0
Do you wish to preform another calculation (Enter y for yes):n


## Programming Exercises 

Question 1: Bug Colector - a bug collector collects bugs every day for five days. Write a program that keeps a running total of the number of bugs collected during the five days. The loop should ask for the number of bugs collected each day, and when the loop is finished, the program should display the total number of bugs collected. 

In [3]:
#create variable to control 
keep_going = 'y'

# Collect inputs and give meaning to y 
while keep_going == 'y':
    #collect number of bugs on day 1 
    day1 = float(input('Enter number of bugs collected on Day 1: '))
    #collect number of bugs on day 2 
    day2 = float(input('Enter number of bugs collected on Day 2: '))
    #collect number of bugs on day 3 
    day3 = float(input('Enter number of bugs collected on Day 3: '))
    #collect number of bugs on day 4 
    day4 = float(input('Enter number of bugs collected on Day 4: '))
    #collect number of bugs on Day 5 
    day5 = float(input('Enter number of bugs collected on Day 5: '))
    
    #Calculate the number of bugs collected 
    total_bugs = day1 + day2 + day3 + day4 + day5 
    
    #display total bugs 
    print(total_bugs)
    
    #ask if user would like to collect again 
    keep_going = input('Do you wish to collect ' + 'again next week?(Enter Y for Yes)')

Enter number of bugs collected on Day 1: 15
Enter number of bugs collected on Day 2: 22
Enter number of bugs collected on Day 3: 53
Enter number of bugs collected on Day 4: 65
Enter number of bugs collected on Day 5: 104
259.0
Do you wish to collect again next week?(Enter Y for Yes)y
Enter number of bugs collected on Day 1: 10
Enter number of bugs collected on Day 2: 5
Enter number of bugs collected on Day 3: 46
Enter number of bugs collected on Day 4: 52
Enter number of bugs collected on Day 5: 2
115.0
Do you wish to collect again next week?(Enter Y for Yes)n


Question 4: Distance Traveled - The distance a vehicle travels can be calculated as follows: 

distance = speed * time 

For example, if a train travels 40 miles per hour for three hours, the distance traveled is 120 miles. Write a program that asks the user for the speed of a vehicle ( in miles per hour) and the number of hours it has traveled. It should then use a loop to display the distance the vehicle has traveled for each hour of the time period. 

In [4]:
#create variable to control loop 
keep_going = 'y'

#collect inputs of speed and hours 
while keep_going == 'y':
    #Collect speed in MPH 
    speed = float(input('Enter Speed in MPH: '))
    #Collect time driving in hours 
    time = int(input('Enter Time in Hours: '))
    
    #Calculate distance with the equation distance = speed * time 
    distance = speed * time 
    
    #Display distance 
    print(distance)
         
    #ask if user would like to calculate again
    keep_going = input('Do you wish to calculate' + ' again?(Enter y for Yes)')
    

Enter Speed in MPH: 85
Enter Time in Hours: 7
595.0
Do you wish to calculate again?(Enter y for Yes)y
Enter Speed in MPH: 75
Enter Time in Hours: 13
975.0
Do you wish to calculate again?(Enter y for Yes)n


Question 12: Calculating the Factorial of a Number - Write a program that lets the user enter a nonnegative integer then uses a loop to calculate the factorial of that number. Display the factorial. 


In [5]:
#enter nonnegative number 
def main():
    #get a nonnegative number from the user 
    number = int(input('Enter Nonnegative Number: '))
    
    #find factorial of the number 
    fact = factorial(number)
    
    #display factorial 
    if number < 0:
        print('')
    else:
        print('The factorial of', number, 'is', fact)

        
#define factorial
def factorial(num):
    #create if-else statement
    if num < 0: 
        print('E - Must use Positive Number') 
    elif num == 0: 
        return 1
    else:
        return num * factorial(num -1) 
        

#create variable to control  
keep_going = 'y'

#collect inputs  
while keep_going == 'y':
    
#Display distance 
    main()
         
    #ask if user would like to calculate again
    keep_going = input('Do you wish to calculate' + ' again?(Enter y for Yes)')


Enter Nonnegative Number: 5
The factorial of 5 is 120
Do you wish to calculate again?(Enter y for Yes)y
Enter Nonnegative Number: 4
The factorial of 4 is 24
Do you wish to calculate again?(Enter y for Yes)y
Enter Nonnegative Number: -5
E - Must use Positive Number

Do you wish to calculate again?(Enter y for Yes)y
Enter Nonnegative Number: 0
The factorial of 0 is 1
Do you wish to calculate again?(Enter y for Yes)n


Question 14: Write a program that uses nested loops to draw the pattern shown in the book 

In [6]:
#setting number of rows
rows = 7
for i in range(rows + 1, 0, -1):
    #nested loop reversed
    for j in range(0, i -1):
        #display "#" symbol and space between each symbol 
        print("#", end = ' ')
    #symbol after each row 
    print(" ")


# # # # # # #  
# # # # # #  
# # # # #  
# # # #  
# # #  
# #  
#  
 
